File: judge-api/src/main/kotlin/io/leonfoliveira/judge/api/controller/dto/response/ContestPrivateResponseDTO.kt
package io.leonfoliveira.judge.api.controller.dto.response

import io.leonfoliveira.judge.core.domain.entity.Contest
import io.leonfoliveira.judge.core.domain.enumerate.Language
import java.time.LocalDateTime

class ContestPrivateResponseDTO(
    val id: Int,
    val title: String,
    val languages: List<Language>,
    val startAt: LocalDateTime,
    val endAt: LocalDateTime,
    val members: List<MemberResponseDTO>,
    val problems: List<ProblemPrivateResponseDTO>,
)

fun Contest.toPrivateResponseDTO(): ContestPrivateResponseDTO {
    return ContestPrivateResponseDTO(
        id = this.id,
        title = this.title,
        languages = this.languages,
        startAt = this.startAt,
        endAt = this.endAt,
        members = this.members.map { it.toPrivateResponseDTO() },
        problems = this.problems.map { it.toPrivateResponseDTO() },
    )
}

File: judge-web/src/core/repository/dto/response/ContestPrivateResponseDTO.ts
import { Language } from "@/core/domain/enumerate/Language";
import { MemberResponseDTO } from "@/core/repository/dto/response/MemberResponseDTO";
import { ProblemPrivateResponseDTO } from "@/core/repository/dto/response/ProblemPrivateResponseDTO";

export type ContestPrivateResponseDTO = {
  id: number;
  title: string;
  languages: Language[];
  startAt: string;
  endAt: string;
  members: MemberResponseDTO[];
  problems: ProblemPrivateResponseDTO[];
};
